plugins/modules/s3_bucket.py

```diff
@@ -1232,9 +1232,11 @@ def create_or_update_bucket(s3_client, module: AnsibleAWSModule):
     """
     name = module.params.get("name")
     object_lock_enabled = module.params.get("object_lock_enabled")
+    location = module.region
```

Comment:
I think it would make more sense to handle this logic in `module_utils/s3.py`

```
get_s3_location(module):
    if module.params.get("ceph") is True:
        return module.params.get("region")
    return module.region or "us-east-1"
```